base/src/org/eevolution/process/ValuationEffectiveDate.java

```diff
@@ -77,10 +77,38 @@ protected String doIt() throws Exception {
 		pstmt.executeBatch();
 		commitEx();
 		DB.close(pstmt);
-		
-		DB.executeUpdate("UPDATE T_InventoryValue SET Cost = CASE WHEN QtyOnHand <> 0 THEN (CostAmt + CostAmtLL) / QtyOnHand ELSE  0 END  ,  CumulatedAmt = CASE WHEN QtyOnHand <> 0  THEN  CostAmt + CostAmtLL ELSE 0 END ,  DateValue = "
+		for (MAcctSchema acctSchema: acctSchemas) {
+			int no = 0;
+			String whereSeq = " (SELECT MAX(SeqNo) FROM M_CostDetail " + 
+					" WHERE IsReversal='N' AND M_Product_ID=iv.M_Product_ID AND iv.m_Costtype_ID=cd.m_Costtype_ID AND iv.c_AcctSchema_ID=cd.c_AcctSchema_ID and iv.m_CostElement_ID=cd.m_Costelement_ID AND dateacct <=?))";
+			if (acctSchema.getCostingLevel().equals(MAcctSchema.COSTINGLEVEL_Client)) {
+				ArrayList<Object> params = new ArrayList<>();
+				params.add(getDateValue());
+				params.add(getDateValue());
+				params.add(getDateValue());
+				params.add(getAD_PInstance_ID());
+				StringBuffer update1 = new StringBuffer( "update t_Inventoryvalue iv ")
+						.append(" set costamt = (select coalesce(currentcostprice,0) from rv_m_transaction_costing cd where iv.m_Product_ID=cd.m_Product_ID AND cd.seqno= ")
+						.append(whereSeq);
+				update1.append(", costamtll = (select coalesce(currentcostpriceLL) from rv_m_transaction_costing cd where iv.m_Product_ID=cd.m_Product_ID AND cd.seqno= ")
```

Comment:
Can change it?
<pre>
set costamt = (select coalesce(currentcostprice,0) from rv_m_transaction_costing cd where iv.m_Product_ID=cd.m_Product_ID AND cd.seqno= 
</pre>

All SQL for ADempiere is better if you use SELECT with uppercase also the column like seqno is better if you can call it like SeqNo